gaia-sdk-javascript/src/api/IdentityOp.ts

```diff
@@ -9,9 +13,37 @@ export class IdentityOp {
         this.client = client;
     }
 
-    public import(identityId?: string): Observable<Blob> | undefined {
-        console.log("TODO: implement import identity");
-        return undefined;
+    public import(tenantId: string, identityName: string, content: Blob, override: boolean = false, identityId?: string): Observable<IIdentityImported> {
+        let file = content as File;
+
+        if (!file.name) {
+            throw new Error('Import identity failed: Selected file is not a valid identity');
+        }
+
+        return from(new DataRef(`gaia://${tenantId}/identities/`, this.client)
```

Comment:
Da musst du mit RX arbeiten und nicht mit Promises.

Ein RX Observable wird erst dann konsumiert wenn jemand subscribe (o.ä.) aufruft.
Eine Promise wird zwar async aber sofort konsumiert.

Was du also willst ist whs ca so etwas:
`new DataRef(`gaia://${tenantId}/identities/`, this.client)
.add(file.name, content, override)
.pipe(mergeMap(dataRef => from(this.importIdentity(dataRef.getUri(), tenantId, identityName, override, identityId).catch(reason => {
                throw new Error('Identity Upload failed: ' + reason);
            }))))
`